stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -201,7 +201,7 @@ internal class PaymentSheetViewModel internal constructor(
                 eventReporter.onPaymentFailure(selection.value)
 
                 val paymentIntent = paymentIntentResult.intent
-                onApiError(paymentIntent.lastPaymentError?.message)
+                onApiError(paymentIntentResult.failureMessage)
```

Comment:
I might need some help coming up with good test cases for this where the lastPaymentError should be used instead of failure message.  Right now in the DefaultPaymentFlowResultProcessor it is using the PaymentFlowFailureMessageFactory to format the api message into a user readable one.

Reply:
Oh nice catch, I think this works.